src/NATS.Client.KeyValueStore/NatsKVOpts.cs

```diff
@@ -36,6 +36,11 @@ public record NatsKVWatchOpts
     /// Meta only to get the metadata of the entries
     /// </summary>
     public bool MetaOnly { get; init; } = false;
+
+    /// <summary>
+    /// Async function called when the enumerator reaches the end of data. Return True to break the async enumeration, False to allow the enumeration to continue.
```

Comment:
The name and description does not make sense to me... What is the "end of data"? I can see from the code that it is NumPending==0, but the name/description do not indicate that.

Also, why does it need to be a callback? Would it be better to make it a Boolean `StopWhenNumPendingZero` or something along those lines?